compiler/rustc_const_eval/src/check_consts/check.rs

```diff
@@ -865,35 +910,7 @@ impl<'tcx> Visitor<'tcx> for Checker<'_, 'tcx> {
                     return;
                 }
 
-                let mut err_span = self.span;
-                let ty_of_dropped_place = dropped_place.ty(self.body, self.tcx).ty;
-
-                let ty_needs_non_const_drop =
-                    qualifs::NeedsNonConstDrop::in_any_value_of_ty(self.ccx, ty_of_dropped_place);
```

Comment:
Seems like previously, we always checked `in_any_value_of_ty` before checking the `qualifs`. Any idea why we did that? Your new logic does not do this any more. Might just be a perf thing?

Reply:
Yeah, I think maybe perf? I'll check perf anyways, but I expect it to not matter.

Reply:
Actually, I have no idea why this would affect perf. 

We call `in_any_value_of_ty` as soon as we call `self.qualifs.needs_drop`.